### PR TITLE
[CAEP] proposal for failure domain

### PR DESCRIPTION
**Pull request by jzhoucliqr**
**What this PR does / why we need it**:
Proposal for failure domain design based on community discussions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Review by yastij — APPROVED** [review]
/approve

**Comment by yastij** [comment]
/assign @randomvariable 

for lgtm

File: docs/proposal/20201103-failure-domain.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
+# CAPV ControlPlane Failure Domain
```
Comment (jzhoucliqr) [inline docs/proposal/20201103-failure-domain.md:1]: @yastij mdlint doesn't like the proposal template to have title after the status. So I'm adding this hack to put the line 4-20 within a code block, and move H1 to first line.

**Review by yastij — COMMENTED** [review]
/lgtm

**Pull request merged**